st2client/st2client/commands/rbac.py

```diff
@@ -0,0 +1,162 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from st2client.formatters import table
+from st2client.commands import resource
+from st2client.models.rbac import Role
+from st2client.models.rbac import UserRoleAssignment
+
+__all__ = [
+    'RoleBranch',
+    'RoleAssignmentBranch'
+]
+
+ROLE_ATTRIBUTE_DISPLAY_ORDER = ['id', 'name', 'system', 'permission_grants']
+ROLE_ASSIGNMENT_ATTRIBUTE_DISPLAY_ORDER = ['id', 'role', 'user', 'is_remote', 'description']
+
+
+class RoleBranch(resource.ResourceBranch):
+    def __init__(self, description, app, subparsers, parent_parser=None):
+        super(RoleBranch, self).__init__(
+            Role, description, app, subparsers,
+            parent_parser=parent_parser,
+            read_only=True,
+            commands={
+                'list': RoleListCommand,
+                'get': RoleGetCommand
+            })
+
+
+class RoleListCommand(resource.ResourceCommand):
+    display_attributes = ['id', 'name', 'system', 'description']
+    attribute_display_order = ROLE_ATTRIBUTE_DISPLAY_ORDER
+
+    def __init__(self, resource, *args, **kwargs):
+        super(RoleListCommand, self).__init__(
+            resource, 'list', 'Get the list of the  %s.' %
+            resource.get_plural_display_name().lower(),
+            *args, **kwargs)
+
+        self.group = self.parser.add_mutually_exclusive_group()
+
+        # Filter options
+        self.group.add_argument('-s', '--system', action='store_true',
+                                help='Only display system roles.')
+
+        # Display options
+        self.parser.add_argument('-a', '--attr', nargs='+',
+                                 default=self.display_attributes,
+                                 help=('List of attributes to include in the '
+                                       'output. "all" will return all '
+                                       'attributes.'))
+        self.parser.add_argument('-w', '--width', nargs='+', type=int,
+                                 default=None,
+                                 help=('Set the width of columns in output.'))
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        # Filtering options
+        if args.system:
```

Comment:
It's a boolean filter which allows user to only display system roles.

`if args.system is None` would also work and perhaps be better, but right now, we only have positive flag and filter in most places so I went with that.

If we also wanted to support inverse operation (only display non-system roles), I would need to change it to `is None` check (and also change how flags work).